### PR TITLE
[FEATURE: UIPT-23] - Shop By Category

### PR DESCRIPTION
**Pull request by andriiveremchuk**

### CODE REVIEW

File: src/app/shared/shop-by-category/shop-by-category.component.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    this.currentPath = this.route.snapshot.url[0].path;
+    this.store.dispatch(new LoadProducts());
+
+    this.filterGenderSub = this.store.select(getFilteredProducts).subscribe(items => this.filteredItems = items);
+  }
+
+  public ngAfterViewChecked(): void {
+    this.currentPath === 'women' || this.currentPath === 'men'
+    ? this.store.dispatch(new FilterByGender(this.currentPath))
+    : this.store.dispatch(new FilterBySeason(this.currentPath));
```
Comment (AlexLarusso) [inline src/app/shared/shop-by-category/shop-by-category.component.ts]: Please align:

```
statement ?
  condition1 :
  condition2;
```

File: src/app/shared/shop-by-category/shop-by-category.scss [hunk 2]
```diff
@@ -0,0 +1,64 @@
+@import '../../styles/variables';
+
+
```
Comment (AlexLarusso) [inline src/app/shared/shop-by-category/shop-by-category.scss]: remove useless line

File: src/app/store/actions/products.action.ts [hunk 3]
```diff
@@ -15,7 +17,21 @@ export class SetProducts implements Action {
   public static readonly TYPE = ProductsActionTypes.SetProducts;
   public type = SetProducts.TYPE;
 
-  constructor(public payload: Array<IProductShortInfo>) { }
+  constructor(public payload: any) { }
```
Comment (AlexLarusso) [inline src/app/store/actions/products.action.ts]: 1) Do you use payload for this action? If no, please remove
2) Constructor should be before properties

Check everywhere

**Comment by AlexLarusso** [comment]
Fix conflicts

File: src/app/shared/shop-by-category/shop-by-category.component.ts (hunk 1 above)
Comment (andriiveremchuk) [reply src/app/shared/shop-by-category/shop-by-category.component.ts]: Isn't this case appropriate aswell?
image

File: src/app/shared/shop-by-category/shop-by-category.component.ts [hunk 4]
```diff
@@ -39,9 +39,9 @@ export class ShopByCategoryComponent implements OnInit, OnDestroy, AfterViewChec
   }
 
   public ngAfterViewChecked(): void {
-    this.currentPath === 'women' || this.currentPath === 'men'
-    ? this.store.dispatch(new FilterByGender(this.currentPath))
-    : this.store.dispatch(new FilterBySeason(this.currentPath));
+    this.currentPath === 'women' || this.currentPath === 'men' ?
+    this.store.dispatch(new FilterByGender(this.currentPath)) :
```
Comment (AlexLarusso) [inline src/app/shared/shop-by-category/shop-by-category.component.ts]: 2 more spaces

```
statement ?
  condition1 :
  condition2
```

Comment (andriiveremchuk) [reply src/app/shared/shop-by-category/shop-by-category.component.ts]: Sorry, my mistake

**Pull request merged**